Vungle/VungleRouter.m

```diff
@@ -299,30 +302,28 @@ - (void)requestBannerAdWithPlacementID:(NSString *)placementID
                 [self.bannerDelegates setObject:delegate forKey:placementID];
             }
 
-            if (needRequestAd) {
-                NSError *error = nil;
-                if (CGSizeEqualToSize(size, kVNGMRECSize)) {
-                    if ([[VungleSDK sharedSDK] loadPlacementWithID:placementID error:&error]) {
-                        MPLogInfo(@"Vungle: Start to load an ad for Placement ID :%@", placementID);
-                    } else {
-                        [self requestBannerAdFailedWithError:error
-                                                 placementID:placementID
-                                                    delegate:delegate];
-                    }
+            NSError *error = nil;
+            if (CGSizeEqualToSize(size, kVNGMRECSize)) {
+                if ([[VungleSDK sharedSDK] loadPlacementWithID:placementID error:&error]) {
+                    MPLogInfo(@"Vungle: Start to load an ad for Placement ID :%@", placementID);
                 } else {
-                    if ([[VungleSDK sharedSDK] loadPlacementWithID:placementID withSize:[self getVungleBannerAdSizeType:size] error:&error]) {
-                        MPLogInfo(@"Vungle: Start to load an ad for Placement ID :%@", placementID);
-                    } else {
-                        if (!error) {
-                            NSString *errorMessage = [NSString stringWithFormat:@"Vungle: Unable to load an ad for Placement ID: %@.", placementID];
-                            error = [NSError errorWithCode:MOPUBErrorAdapterFailedToLoadAd
-                                      localizedDescription:errorMessage];
-                        }
-                        [self requestBannerAdFailedWithError:error
-                                                 placementID:placementID
-                                                    delegate:delegate];
-                        MPLogError(@"%@", error);
+                    [self requestBannerAdFailedWithError:error
```

Comment:
Does there need to be a check on `error` before sending it in the `requestBannerAdFailedWithError: placementID: delegate:` method?

Reply:
Looking at lines 318 - 322 and wondering if the same should be done prior to this method call. Thanks!

Reply:
Also - should we include a `MPLogInfo(@"");` line after this method?

Reply:
@crbinsf Hi Clarke, thanks a lot to point out this issue. In `- (void)requestBannerAdFailedWithError:placementID:delegate:` method, we will also check the error is nil and also log the error `MPLogError(@"")`, so the lines `318-322` and line `326` are redundant. I will remove these redundant lines. 🙂

Reply:
@crbinsf Removed, thanks. :)